charmcraft/commands/pack.py

```diff
@@ -114,18 +88,14 @@ def run(self, parsed_args):
                 "Invalid bundle config; missing a 'name' field indicating the bundle's name in "
                 "file '{}'.".format(bundle_filepath))
 
-        charmcraft_filepath = dirpath / 'charmcraft.yaml'
-        charmcraft_config = load_yaml(charmcraft_filepath)
-        if charmcraft_config is None:
-            raise CommandError(
-                "Missing or invalid charmcraft file: '{}'.".format(charmcraft_filepath))
-        if charmcraft_config.get('type') != 'bundle':
+        # so far 'pack' works for bundles only (later this will operate also on charms)
+        if self.config.type != 'bundle':
             raise CommandError(
-                "Invalid charmcraft config; 'type' must be 'bundle' in file '{}'."
-                .format(charmcraft_filepath))
+                "Bad config; charmcraft.yaml missing or with an invalid 'type' field: "
```

Comment:
Consider splitting up missing vs. unsupported type.

Reply:
Ok, will make this a more specific requirement.